src/components/molecules/table-fileds-filter/index.tsx

```diff
@@ -0,0 +1,202 @@
+  const { fields, onBlur, selectedFields } = props
+
+  const contentRef = useRef()
+  const [open, setOpen] = useState(false)
+  // local copy of selected filters which is synced with the container list on blur
+  const [currentlySelected, setCurrentlySelected] = useState<string[]>([])
+
+  const onTriggerClick = () => {
+    setOpen(true)
+  }
+
+  const toggleCheck = (id: string) => {
+    if (currentlySelected.includes(id)) {
+      setCurrentlySelected(currentlySelected.filter((f) => f !== id))
+    } else {
+      setCurrentlySelected([...currentlySelected, id])
+    }
+  }
+
+  useEffect(() => {
+    if (open) {
+      setCurrentlySelected(selectedFields)
+    }
+  }, [open, selectedFields])
+
+  useEffect(() => {
+    if (!open) {
+      onBlur(currentlySelected)
+    }
+  }, [open])
+
+  // close dropdown "manually" on click outside the menu
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (!contentRef.current?.contains(event.target)) {
+        setOpen(false)
+      }
+    }
+    document.addEventListener("mousedown", handleClickOutside)
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside)
+    }
+  }, [contentRef])
+
+  return (
+    <DropdownMenu.Root open={open}>
+      <DropdownMenu.Trigger>
+        <Button
+          onClick={onTriggerClick}
+          variant="secondary"
+          className="rounded rounded-lg h-[32px] px-3 text-small font-semibold text-grey-90"
+        >
+          <span className="flex whitespace-nowrap items-center gap-1">
+            Add fields <PlusIcon size={14} />
+          </span>
+        </Button>
+      </DropdownMenu.Trigger>
+
+      <DropdownMenu.Content
+        ref={contentRef}
+        className="w-[240px] bg-white shadow rounded-xl p-2"
+      >
+        {fields.map((f) => (
+          <FieldMenuItem
+            key={f.id}
+            field={f}
+            onChange={() => toggleCheck(f.id)}
+            checked={currentlySelected.includes(f.id)}
+          />
+        ))}
+      </DropdownMenu.Content>
+    </DropdownMenu.Root>
+  )
+}
+
+/**
+ * Table fields filter root container.
+ */
+function TableFieldsFilters(props: TableFieldsFilterProps) {
+  const { fields, onChange } = props
+
+  const [selectedFields, setSelectedFields] = useState<Field["id"][]>([])
+
+  useEffect(() => {
+    onChange(selectedFields)
+  }, [selectedFields])
+
+  const removeSelected = (id: string) => {
+    setSelectedFields(selectedFields.filter((f) => f !== id))
+  }
+
+  const _selected = [...selectedFields]
+  _selected.sort((a, b) => a.localeCompare(b))
+
+  const visibleFields = _selected.map((id) => fields.find((f) => f.id === id))
+
+  return (
+    <div className="flex items-center gap-2">
+      <span className="text-small font-semibold text-gray-500 whitespace-nowrap">
+        Currently editing these fields:
+      </span>
+
+      <div className="flex gap-1 overflow-x-auto no-scrollbar::-webkit-scrollbar">
```

Comment:
```suggestion
      <div className="flex gap-1 overflow-x-auto no-scrollbar">
```
Also requires adding the utility classes as mentioned in another comment

Reply:
I was experimenting with scrollbar hiding and forgot to remove this class but I'm not sure whether it's a good UX practice if we hide the scrollbar though.